### PR TITLE
[GPU] fix priorbox legacy code in CreateOp() and allow to use max_size is null

### PR DESCRIPTION
**Pull request by wilson-seok**
### Details:
 - fix priorbox legacy code in CreateOp()
 - allow to use max_size is null

### Tickets:
 - 101235

### CODE REVIEW

File: src/plugins/intel_gpu/src/plugin/ops/prior_box.cpp [hunk 1]
```diff
@@ -172,18 +184,23 @@ static void CreatePriorBoxOp(Program& p, const std::shared_ptr<ngraph::op::v8::P
...
         const auto height = output_size[1];
         const cldnn::tensor output_size_tensor{cldnn::spatial(width, height)};
 
+        cldnn::tensor img_size_tensor{};
+        if (image_size_constant) {
         const auto image_size = image_size_constant->cast_vector<int64_t>();
         const auto image_width = image_size[0];
         const auto image_height = image_size[1];
-        const cldnn::tensor img_size_tensor{cldnn::spatial(image_width, image_height)};
+        img_size_tensor = (cldnn::tensor) cldnn::spatial(image_width, image_height);
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/plugin/ops/prior_box.cpp:202]: Please add proper indents.

Comment (wilson-seok) [reply src/plugins/intel_gpu/src/plugin/ops/prior_box.cpp:204]: Added. Thanks!

File: src/plugins/intel_gpu/src/plugin/ops/prior_box.cpp [hunk 2]
```diff
@@ -111,18 +111,31 @@ static void CreatePriorBoxOp(Program& p, const std::shared_ptr<ngraph::op::v0::P
...
-        auto hdim = img_shape.at(img_shape.size()-2);
+        const auto output_size = output_size_constant->cast_vector<int64_t>();
+        const auto width = output_size[0];
+        const auto height = output_size[1];
+        const cldnn::tensor output_size_tensor{cldnn::spatial(width, height)};
+
+        cldnn::tensor img_size_tensor{};
+        // When image size is constant, set the value for primitive construction. Others don't have to set it. It will be determined in execute_impl time.
+        if (image_size_constant) {
+        const auto image_size = image_size_constant->cast_vector<int64_t>();
```
Comment (andrew-k-park) [inline src/plugins/intel_gpu/src/plugin/ops/prior_box.cpp:129]: Please add proper indents here too.

Comment (wilson-seok) [reply src/plugins/intel_gpu/src/plugin/ops/prior_box.cpp:129]: My mistake. Fixed. Thanks!

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/prior_box_ref.cl [hunk 3]
```diff
@@ -121,52 +121,56 @@ KERNEL(ref)
...
-    for (uint ms_idx = 0; ms_idx < MIN_SIZE_SIZE; ++ms_idx) {
-        box_width = MIN_SIZE[ms_idx] * 0.5f;
-        box_height = MIN_SIZE[ms_idx] * 0.5f;
-        FUNC_CALL(calculate_data)(center_x, center_y, box_width, box_height, false, &out_index, output);
-        #ifdef MIN_MAX_ASPECT_RATIO_ORDER
-            if (MAX_SIZE_SIZE > ms_idx) {
-                box_width = box_height = sqrt(MIN_SIZE[ms_idx] * MAX_SIZE[ms_idx]) * 0.5f;
-                FUNC_CALL(calculate_data)(center_x, center_y, box_width, box_height, false, &out_index, output);
+    // Explicitly check MIN_SIZE has value to avoid seg fault duing opencl build
+    if (MIN_SIZE_SIZE > 0) {
```
Comment (e-ddykim) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/prior_box_ref.cl:125]: This condition check is unnecessary. It is checked by the next `for` statement.

Comment (vladimir-paramuzov) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/prior_box_ref.cl:125]: Why those are blank? I think it can always default to 0, isn't it?

Comment (e-ddykim) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/prior_box_ref.cl:125]: When `MAX_SIZE_SIZE` is 0, `MAX_SIZE` is defined as `#define MAX_SIZE (float []){  }`
In this case, `MAX_SIZE[ms_idx]` causes programs to crash.
But, if this code is under a condition `MAX_SIZE_SIZE > 0`, it looks like that this code is excluded before compilation.

**Review by e-ddykim — APPROVED** [review]

**Review by vladimir-paramuzov — APPROVED** [review]

**Pull request merged**